ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/logic.j2

```diff
@@ -1,9 +1,35 @@
 {% for connected_endpoints_key in connected_endpoints_keys | arista.avd.convert_dicts('key') %}
 {%     set switch_connected_endpoints = hostvars[inventory_hostname][connected_endpoints_key.key] | arista.avd.convert_dicts('name') | arista.avd.default([]) %}
-{%     do connected_endpoints.update({ connected_endpoints_key.key: {} }) %}
 {%     for connected_endpoint in switch_connected_endpoints %}
-{%         if connected_endpoint.adapters | selectattr('switches','arista.avd.contains',inventory_hostname) | length > 0 %}
-{%             do connected_endpoints[connected_endpoints_key.key].update({connected_endpoint.name: connected_endpoint}) %}
+{%         if connected_endpoint.adapters | selectattr('switches', 'arista.avd.contains', inventory_hostname) | length > 0 %}
+{%             do connected_endpoint.update({"type": connected_endpoints_key.type}) %}
+{%             do connected_endpoints.append(connected_endpoint) %}
 {%         endif %}
 {%     endfor %}
 {% endfor %}
+{% if network_ports is arista.avd.defined %}
+{%     for network_port_item in network_ports %}
+{%         for switch_regex in network_port_item.switches | arista.avd.default([]) %}
+{%             if switch_regex[-1] != "$" %}
+{#                 The match test is built on Python re.match which tests form the beginning of the string #}
+{#                 Since the user would not expect "DC1-LEAF1" to also match "DC-LEAF11" we will force $ at the end #}
```

Comment:
What if the use does expect this? I am not a huge fan of this and would rather explicitly document some example instead as this is kind of "breaking" regex for people used to them

Reply:
I think if we're to keep this behaviour it would need to be documented.  What for example if the user used DC1-LEAF to match all leaves in DC1.  They would have to specify DC1-LEAF.* (for e.g.) which IMO is nicer as it's more explicit, but agree with @gmuloc - it may be unexpected for some.

Reply:
We do not support regex today. This is a completely new data model, so users would not have anything set. But the point is that for users writing a regular hostname, they would expect that to match that single hostname. And with default re.match it does not.

Reply:
It is mentioned in the documentation that we expect the regex to match the full hostname.